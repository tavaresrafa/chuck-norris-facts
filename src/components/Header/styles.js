import styled, { keyframes } from 'styled-components';

export const Container = styled.header`
  width: 100%;
  display: flex;
  justify-content: center;
  margin-bottom: 4rem;
`;

// Logo Animation
const shakeAnimation = keyframes`
  0% { transform: rotate(0deg); }
  10% { transform: rotate(1deg); }
  20% { transform: rotate(0deg); }
  30% { transform: rotate(-1deg); }
  40% { transform: rotate(0deg); }
  50% { transform: rotate(1deg); }
  60% { transform: rotate(0deg); }
  70% { transform: rotate(-1deg); }
  80% { transform: rotate(0deg); }
  90% { transform: rotate(1deg); }
  100% { transform: rotate(0deg); }
`;

export const Logo = styled.picture`
  max-width: 100%;

  @media (min-width: 360px) {
    max-width: 36rem;
  }

  :hover {
    animation-name: ${shakeAnimation};
    animation-duration: 0.4s;
  }
`;
