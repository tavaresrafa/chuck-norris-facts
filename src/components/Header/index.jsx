import React from 'react';
import * as S from './styles';
import Image from '../../assets/img/chucknorris_logo.png';

const Header = () => (
  <S.Container>
    <S.Logo>
      <source
        media="(min-width: 800px)"
        srcSet={`${Image} 1x, ${Image} 2x`}
      />
      <source
        media="(min-width: 400px)"
        srcSet={`${Image} 1x, ${Image} 2x`}
      />
      <source srcSet={`${Image} 1x, ${Image} 2x`} />
      <img src={Image} alt="Logo do site." />
    </S.Logo>
  </S.Container>
)

export default Header;