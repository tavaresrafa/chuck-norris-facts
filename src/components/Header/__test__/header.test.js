import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../../Header';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Header />, div);
});

// SNAPSHOTS
it('matches snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
});