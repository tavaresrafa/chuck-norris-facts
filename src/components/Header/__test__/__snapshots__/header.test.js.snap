// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="sc-AxjAm djeBuX"
>
  <picture
    className="sc-AxirZ dRTkRp"
  >
    <source
      media="(min-width: 800px)"
      srcSet="chucknorris_logo.png 1x, chucknorris_logo.png 2x"
    />
    <source
      media="(min-width: 400px)"
      srcSet="chucknorris_logo.png 1x, chucknorris_logo.png 2x"
    />
    <source
      srcSet="chucknorris_logo.png 1x, chucknorris_logo.png 2x"
    />
    <img
      alt="Logo do site."
      src="chucknorris_logo.png"
    />
  </picture>
</header>
`;
